iothub_client/samples/pnp/pnp_simple_thermostat/pnp_simple_thermostat.c

```diff
@@ -0,0 +1,484 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// This sample implements a PnP based thermostat.  This demonstrates a *relatively* simple PnP device
+// that does only acts as a thermostat and does not have additional components.  
+
+// The DigitalTwin Definition Language document describing the component implemented in this sample
+// is available at https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/Thermostat.json.
+
+// Standard C header files
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+// IoTHub Device Client and IoT core utility related header files
+#include "iothub.h"
+#include "iothub_device_client.h"
+#include "iothub_message.h"
+#include "azure_c_shared_utility/threadapi.h"
+#include "azure_c_shared_utility/xlogging.h"
+
+// JSON parser library
+#include "parson.h"
+
+// pnp_device_client_helpers.h is part of the PnP sample code and helps in 
+// creation of the IOTHUB_DEVICE_CLIENT_HANDLE
+#include "pnp_device_client_helpers.h"
+
+// Environment variable used to specify this application's connection string
+static const char g_ConnectionStringEnvironmentVariable[] = "IOTHUB_DEVICE_CONNECTION_STRING";
+
+// Amount of time to sleep between sending telemetry to Hub, in milliseconds.  Set to 1 minute.
+static unsigned int g_sleepBetweenTelemetrySends = 60 * 1000;
+
+// Whether verbose tracing at the IoTHub client is enabled or not.
+static bool g_hubClientTraceEnabled = true;
+
+// This device's PnP ModelId.
+static const char g_ModelId[] = "dtmi:com:example:Thermostat;1";
+
+// JSON fields from desired property to retrieve.
+static const char PnP_JsonPropertyVersion[] = "$version";
+static const char PnP_JsonTargetTemperature[] = "targetTemperature";
+
+// Name of command this component supports to get report information
+static const char g_GetMinMaxReport[] = "getMaxMinReport";
+// Name of json field to parse for "since". (Note the "since" is assumed as only field of the value)
+static const char g_SinceJsonCommandSetting[] = "commandRequest.value";
```

Comment:
per BUG 7595911. commands should not expect the `commandRequest` envelope

Reply:
Agreed.  This is currently what the hub is sending us.  Once the hub is rev'd to stop doing that, the device code will react.  Other SDKs are tracking this pending work item, too.